src/components/GlobalDataProvider.tsx

```diff
@@ -90,7 +99,7 @@ const GlobalDataProvider = ({ children }: { children: ReactElement }) => {
       clearInterval(congestionInterval);
       setAoCongested(false);
     };
-  });
+  }, [aoCuUrl, arioProcessId, arweave.blocks, setAoCongested, setBlockHeight]);
```

Comment:
this could go into a useQuery hook with with a staleTime of TWO_MINUTES and then have a useEffect listen to the query result to show toast on error, then you can reuse the blockheight data cache elsewhere.

I believe it might even be possible to define an error handler in the QueryClientProvider context (or the query client itself) to manage error handling on individual queries, eg for setting AO network congestion or other error notification. Even sentry error capture.

Reply:
Added a Task (PE-7521) to JIRA to look at this in the future: